### PR TITLE
chore(ci): clean up release and promote workflows

### PR DESCRIPTION
**Pull request by jonkoops**
Cleans up the release and promote workflows so that secrets are only being passed where used, as well as removing some redundant code.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -17,27 +17,22 @@ jobs:
...
+          # Fetch all history for all branches and tags, which is needed for the release script.
           fetch-depth: 0
-          # Needs to be an admin token to get around branch protection.
+          # Pass in an administrator token to get around branch protection.
           token: ${{ secrets.GH_TOKEN_REDALLEN }}
 
-      # Fetches all tags from 'origin' and updates local tags, only fetches the latest commit.
-      - name: Fetch tags
-        run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
-
```
Comment (jonkoops) [inline .github/workflows/release.yml:36]: I've remove this as `fetch-depth: 0` already fetches all the tags, see https://github.com/actions/checkout/issues/217#issuecomment-900923602.

**Review by dlabaj — APPROVED** [review]
LGTM

File: .github/workflows/promote.yml [hunk 2]
```diff
@@ -1,23 +1,18 @@
-name: promote
+name: Promote
 on:
   workflow_dispatch:
     inputs:
-      core-version:
-        description: The PatternFly core version
+      release-version:
+        description: The version of PatternFly to promote to.
```
Comment (nicolethoen) [inline .github/workflows/promote.yml:6]: FWIW, the argument in this workflow is the @patternfly/patternfly package version which is updated where ever necessary in the various packages in the patternfly-react monorepo. So it _should_ be the same as the promoted version, but it isn't necessarily always.

File: .github/workflows/release.yml (hunk 1 above)
Comment (nicolethoen) [reply .github/workflows/release.yml:36]: Fetch depth 0 was added (i believe) to make sure we weren't republishing EVERY package in the monorepo each time a single change went into one package. It only republished the relevant packages. I guess we can watch to make sure this doesnt undo that effort now that it's removed.

File: .github/workflows/promote.yml (hunk 2 above)
Comment (jonkoops) [reply .github/workflows/promote.yml]: I understood that as well, but I thought `core-version` was a bit non-descriptive, I've changed this to `patternfly-version` now, with some additional description.

File: .github/workflows/release.yml (hunk 1 above)
Comment (jonkoops) [reply .github/workflows/release.yml:36]: So from what I understand Lerna should take care of this for us, but I can see this issue happening if Lerna does not detect any tags. That perhaps happened before if the `fetch-depth` wasn't set in the action, in which case the default is `1`, which would only fetch the latest commit, and perhaps not all the tags.

The [Lerna docs](https://lerna.js.org/docs/features/distribute-tasks#cicd-examples) link to [an Nx example](https://nx.dev/ci/recipes/set-up/monorepo-ci-github-actions) that seem to operate on the same premise. From what I can gather from community blog posts it seems that `fetch-depth: 1` should be enough for Lerna to do the right thing.

Comment (nicolethoen) [reply .github/workflows/release.yml:36]: Since fetch-depth has 1 as a default value, this should be okay. let's just keep an eye on it.

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**